src/lib/pages/contract-details/components/ContractTop.tsx

```diff
@@ -200,65 +198,74 @@ export const ContractTop = ({
               direction={{ base: "column", md: "row" }}
               gap={{ base: 0, md: 2 }}
             >
-              <Text color="text.dark" variant="body2" fontWeight={500}>
-                Public Contract Name:
+              <Text
+                color="text.dark"
+                minW={32}
+                variant="body2"
+                fontWeight={500}
+              >
+                Public Name
               </Text>
               <Text variant="body2" className="ellipsis">
                 {publicInfo.name}
               </Text>
             </Flex>
           )}
-          {publicInfo?.github && <GitHubLink github={publicInfo?.github} />}
-        </Flex>
-        <Flex
-          gap={{ base: 2, md: 4 }}
-          mt={{ base: 8, md: 0 }}
-          w={{ base: "full", md: "auto" }}
-        >
-          {!isMobile && (
-            <AdminButton
-              contractAddress={contractAddress}
-              admin={contract.admin}
-            />
+          {publicInfo?.github && (
+            <GitHubLink github={publicInfo?.github} hasMinW />
           )}
-
-          <Button
-            variant="outline-primary"
-            w={{ base: "full", md: "auto" }}
-            leftIcon={<CustomIcon name="query" />}
-            onClick={goToQuery}
-            size={{ base: "sm", md: "md" }}
-          >
-            Query
-          </Button>
-          <Button
-            variant="outline-primary"
+        </Flex>
+        <Flex direction="column" gap={4}>
+          <Flex
+            gap={{ base: 2, md: 4 }}
+            mt={{ base: 8, md: 0 }}
             w={{ base: "full", md: "auto" }}
-            leftIcon={<CustomIcon name="execute" />}
-            onClick={goToExecute}
-            size={{ base: "sm", md: "md" }}
           >
-            Execute
-          </Button>
-          {!isMobile && (
-            <Flex>
-              {contractLocalInfo && (
-                <EditContractDetailsModal
-                  contractLocalInfo={contractLocalInfo}
-                  triggerElement={
-                    <IconButton
-                      fontSize="24px"
-                      variant="none"
-                      aria-label="edit"
-                      color="gray.600"
-                      icon={<CustomIcon name="edit" />}
-                    />
-                  }
-                />
-              )}
-              {renderSaveButton()}
-            </Flex>
-          )}
+            {!isMobile && (
+              <AdminButton
+                contractAddress={contractAddress}
+                admin={contract.admin}
+              />
+            )}
+            <Button
+              variant="outline-primary"
+              w={{ base: "full", md: "auto" }}
+              leftIcon={<CustomIcon name="query" />}
+              onClick={goToQuery}
+              size={{ base: "sm", md: "md" }}
+            >
+              Query
+            </Button>
+            <Button
```

Comment:
should we also disable this button on mobile?

Reply:
remain like this, same as current version on prod